### PR TITLE
fix(model): improve conversion and inference examples

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary

- land a focused subset of conversion and inference fixes discovered during broader model validation
- add or update portable examples and documentation for selected model families
- improve the model bridges and shared VLM utilities needed by those flows

## Scope

This PR intentionally merges only the ready, independently useful fixes and examples. It no longer tracks or claims complete validation across every supported model. The broad model inventory and exhaustive per-model checklist have been removed from this PR.

## Follow-up validation

Full-checkpoint conversion, inference, export, round-trip, numerical-parity, and training verification will continue through model verification cards. Each card will capture the model-specific configuration, complete commands, environment, and validation evidence so that coverage can be added incrementally and reviewed independently.

## Validation

- targeted unit tests for the touched conversion, model, and VLM paths
- pre-commit checks
- CI

### CODE REVIEW

File: src/megatron/bridge/models/hybrid/hybrid_provider.py [hunk 1]
```diff
@@ -240,11 +242,19 @@ def _resolve_hybrid_stack_spec(self) -> ModuleSpec:
...
+        elif not isinstance(hybrid_stack_spec, ModuleSpec):
             if len(inspect.signature(hybrid_stack_spec).parameters) > 0:
-                return hybrid_stack_spec(self)
-            return hybrid_stack_spec()
+                hybrid_stack_spec = hybrid_stack_spec(self)
+            else:
+                hybrid_stack_spec = hybrid_stack_spec()
+
+        if self.attention_backend in {AttnBackend.local, "local"}:
+            hybrid_stack_spec = copy.deepcopy(hybrid_stack_spec)
```
Comment (yaoyu-33) [inline src/megatron/bridge/models/hybrid/hybrid_provider.py]: put DotProductAttention import here

File: examples/models/full-model-validation.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Full-model validation inventory
```
Comment (yaoyu-33) [inline examples/models/full-model-validation.md]: it's only verified conversion / inference not full. Also since we fixed few models. Update this table

File: src/megatron/bridge/models/hybrid/hybrid_provider.py (hunk 1 above)
Comment (yaoyu-33) [reply src/megatron/bridge/models/hybrid/hybrid_provider.py]: Addressed in f976fcff8: DotProductAttention is now imported only inside the local-attention branch. I also added persistent coverage for both AttnBackend.local and the string "local" path; the updated Hybrid/GPT-OSS target passed 35/35.

File: examples/models/full-model-validation.md (hunk 2 above)
Comment (yaoyu-33) [reply examples/models/full-model-validation.md]: Addressed in f976fcff8, 4993b7835, and afb4d4ec1: the inventory is scoped to full-checkpoint conversion/inference validation, and PASS is explicitly the six Stage 1 smoke gates (not strict logits/training/model-quality validation). The latest commit adds exact Ministral 3 Base 3B/8B/14B paths, fixes composite tied-weight config semantics, and supports raw completion-style VLM prompts. Clean immutable-commit EOS jobs 5623037/5623038/5623039 each passed 6/6 with three coherent outputs and consistent round-trip configs. The tracker is now synchronized to 33 PASS, 12 BLOCKED, 2 PARTIAL, and 1 DUPLICATE; the other 19 out-of-scope checkpoints remain TODO.
